lib/oli_web/live/collaboration_live/collab_space_view.ex

```diff
@@ -170,15 +177,32 @@ defmodule OliWeb.CollaborationLive.CollabSpaceView do
                 }
                 class="torus-input border-r-0 collab-space__textarea"
               />
-              <div class="flex justify-end">
-                <button
-                  disabled={is_archived?(@collab_space_config.status)}
-                  type="submit"
-                  class="torus-button primary"
-                >Create Post</button>
-              </div>
             </Field>
           </Inputs>
+          <div class="flex justify-end">
+            {#if @is_student and @collab_space_config.anonymous_posting}
+              <Checkbox id="new_post_anonymous_checkbox" field={:anonymous} class="hidden" />
+              <Buttons.button_with_options
+                id="create_post_button"
+                type="submit"
+                disabled={is_archived?(@collab_space_config.status)}
+                options={[
+                  %{
+                    text: "Post anonymously",
+                    on_click:
+                      JS.dispatch("click", to: "#new_post_anonymous_checkbox")
+                      |> JS.dispatch("click", to: "#create_post_button_button")
```

Comment:
Shouldn't this be `#create_post_button`?

Reply:
Yes, the naming may be confusing. The button_with_options component has `id="create_post_button"`, and is composed by two buttons (one that submits the form, other that shows/hides the options). With this JS command we intend to click the button that submits the form (with `id={"#{button_with_options_id}_button"}` that ends up being `id="create_post_button_button"` )